### DISCUSSION TITLE
Global styles, scope styles, themes and variables is possible to implement

### DISCUSSION THREAD

**Discussion by leamandeep** [johanholmerin/style9#96] [Q&A]
I was looking for CSS in JS framework, compiler, etc. for my component library's core package. And luckily after so many successes and failures, I found this library. I'm trying to build somewhat like tailwind and vanilla extract. I found Style9 to be a great tool for accomplishing my requirements.

Just need guidance on how to implement Global styles, scope styles, themes, and variables in style9.
I'm trying to give a config file just like Tailwind to give new values. And  Global styles, scope styles, themes, and variables like vanilla extract. 

Thanks in advance

**Comment by johanholmerin** [comment]
- There's no built in support for global styles, that's best solved with a plain CSS file
- Not sure what you mean by scope styles, the CSS generated by style9 will only apply to the elements you choose
- Theming and variables can be done using CSS custom properties, which you can either define globally in a plain CSS file, or on an element using style9. There's a section about it in the documentation: https://github.com/johanholmerin/style9/blob/master/docs/Usage-guide.md#theming

Hope that helps
